CMakeFilters.cmake

```diff
@@ -71,20 +71,20 @@ if (HDF5_ENABLE_Z_LIB_SUPPORT)
   if (NOT H5_ZLIB_HEADER)
     if (NOT ZLIB_USE_EXTERNAL)
       find_package (ZLIB NAMES ${ZLIB_PACKAGE_NAME}${HDF_PACKAGE_EXT} COMPONENTS static shared)
-      if (NOT ZLIB_FOUND)
+      if (NOT H5_ZLIB_FOUND)
         find_package (ZLIB) # Legacy find
       endif ()
-      if (ZLIB_FOUND)
+      if (H5_ZLIB_FOUND)
```

Comment:
Looks like we'll either need to merge the zlib/szip fixes immediately after this PR or add them here.